ui/pages/reset_email.tsx

```diff
@@ -1,30 +1,90 @@
+import { RecoveryFlow, UpdateRecoveryFlowBody } from "@ory/client";
 import type { NextPage } from "next";
+import { useRouter } from "next/router";
+import { useEffect, useState, useCallback } from "react";
 import React from "react";
+import { handleFlowError } from "../util/handleFlowError";
+import { Flow } from "../components/Flow";
+import { kratos } from "../api/kratos";
 import PageLayout from "../components/PageLayout";
-import { Button, Input } from "@canonical/react-components";
+import { Spinner } from "@canonical/react-components";
+
+const ResetEmail: NextPage = () => {
+  const [flow, setFlow] = useState<RecoveryFlow>();
+
+  // Get ?flow=... from the URL
+  const router = useRouter();
+  const { return_to: returnTo, flow: flowId } = router.query;
+
+  useEffect(() => {
+    // If the router is not ready yet, or we already have a flow, do nothing.
+    if (!router.isReady || flow) {
+      return;
+    }
+
+    // If ?flow=... was in the URL, we fetch it
+    if (flowId) {
+      kratos
+        .getRecoveryFlow({ id: String(flowId) })
+        .then((res) => setFlow(res.data))
+        .catch(handleFlowError(router, "recovery", setFlow));
+      return;
+    }
+
+    // Otherwise we initialize it
+    kratos
+      .createBrowserRecoveryFlow({
+        returnTo: returnTo ? String(returnTo) : undefined,
+      })
+      .then(({ data }) => {
+        if (data.request_url !== undefined) {
+          window.location.href = data.request_url;
```

Comment:
question: Why redirect there? This is the URL of the request that the backend made.

Reply:
Without the redirect the flow id is not appended to the url

Reply:
Doesn't this make the first call (the one done by the backend) useless? Can't the frontend just append the flow_id query param to the URL?